contracts/Voting.sol

```diff
@@ -0,0 +1,157 @@
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+
+// openzeppelin imports
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+
+/// @title Upvoting / Downvoting Contract
+contract Voting {
+  using SafeMath for uint256;
+
+  event ItemVotesUpdated(
+    uint256 indexed itemId,
+    uint256 upvotes,
+    uint256 downvotes,
+    uint256 timestamp
+  );
+
+  event ItemVotesAction(
+    address indexed user,
+    VotesAction indexed action,
+    uint256 indexed itemId,
+    uint256 timestamp
+  );
+
+  enum VotesAction {
+    upvoted,
+    removeUpvoted,
+    downvoted,
+    removeDownvoted
+  }
```

Comment:
I'd rename the actions to `upvote`, `removeUpvote`, `downvote` and `removeDownvote`, to be verbs instead of being in the past. WDYT?

Reply:
Makes sense